File: Models/TaiKhoan.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace WebsiteSmartHome.Models;

[Table("TaiKhoan")]
[Index("Email", Name = "UQ__TaiKhoan__A9D1053492856D42", IsUnique = true)]
[Index("TenTaiKhoan", Name = "UQ__TaiKhoan__B106EAF836A161B9", IsUnique = true)]
public partial class TaiKhoan
{
    [Key]
    public Guid Id { get; set; }

    [StringLength(100)]
    public string TenTaiKhoan { get; set; } = null!;

    [StringLength(255)]
    public string MatKhau { get; set; } = null!;

    [StringLength(255)]
    public string Email { get; set; } = null!;

    [Column(TypeName = "datetime")]
    public DateTime? NgayTao { get; set; }

    [StringLength(20)]
    public string TrangThai { get; set; } = null!;

    [InverseProperty("MaTaiKhoanNavigation")]
    public virtual NguoiDung? NguoiDung { get; set; }
}

File: Frontend/admindoan/src/types/taiKhoan.ts
import { NguoiDungDto } from './auth';

export interface TaiKhoanDto {
  id: string;
  email: string;
  tenTaiKhoan: string;
  matKhau: string;
  ngayTao: Date;
  trangThai: string;
  vaiTro: string;
  maNguoiDung: string;
  MaNguoiDung?: string;
  nguoiDung?: NguoiDungDto;
}

export interface UpdateTaiKhoanDto {
  email: string;
  tenTaiKhoan: string;
  trangThai: string;
}

export interface TaiKhoanCreateDto {
  email: string;
  tenTaiKhoan: string;
  matKhau: string;
  trangThai?: string;
  ngayTao?: Date;
} 